pkg/metrics/metrics.go

```diff
@@ -71,41 +74,69 @@ func NewServerMetrics() *ServerMetrics {
 				},
 				[]string{scheduleLabel},
 			),
+			backupDurationSeconds: prometheus.NewHistogramVec(
+				prometheus.HistogramOpts{
+					Namespace: metricNamespace,
+					Name:      backupDurationSeconds,
+					Help:      "Time taken to complete backup, in seconds",
+					// Split backup observations into buckets based on number of minutes.
+					Buckets: []float64{
+						1 * secondsInMinute,
+						2 * secondsInMinute,
+						3 * secondsInMinute,
+						4 * secondsInMinute,
+						5 * secondsInMinute,
+						6 * secondsInMinute,
+						7 * secondsInMinute,
+						8 * secondsInMinute,
+						9 * secondsInMinute,
+						10 * secondsInMinute,
+					},
+				},
+				[]string{scheduleLabel},
+			),
 		},
 	}
 }
 
-// RegisterAllMetrics registers all prometheus metrics
+// RegisterAllMetrics registers all prometheus metrics.
 func (m *ServerMetrics) RegisterAllMetrics() {
 	for _, pm := range m.metrics {
 		prometheus.MustRegister(pm)
 	}
 }
 
-// SetBackupTarballSizeBytesGauge records the size, in bytes, of a backup tarball
+// SetBackupTarballSizeBytesGauge records the size, in bytes, of a backup tarball.
 func (m *ServerMetrics) SetBackupTarballSizeBytesGauge(backupSchedule string, size int64) {
 	if g, ok := m.metrics[backupTarballSizeBytesGauge].(*prometheus.GaugeVec); ok {
 		g.WithLabelValues(backupSchedule).Set(float64(size))
 	}
 }
 
-// RegisterBackupAttempt records an backup attempt
+// RegisterBackupAttempt records an backup attempt.
 func (m *ServerMetrics) RegisterBackupAttempt(backupSchedule string) {
 	if c, ok := m.metrics[backupAttemptCount].(*prometheus.CounterVec); ok {
 		c.WithLabelValues(backupSchedule).Inc()
 	}
 }
 
-// RegisterBackupSuccess records a successful completion of a backup
+// RegisterBackupSuccess records a successful completion of a backup.
 func (m *ServerMetrics) RegisterBackupSuccess(backupSchedule string) {
 	if c, ok := m.metrics[backupSuccessCount].(*prometheus.CounterVec); ok {
 		c.WithLabelValues(backupSchedule).Inc()
 	}
 }
 
-// RegisterBackupFailed records a failed backup
+// RegisterBackupFailed records a failed backup.
 func (m *ServerMetrics) RegisterBackupFailed(backupSchedule string) {
 	if c, ok := m.metrics[backupFailureCount].(*prometheus.CounterVec); ok {
 		c.WithLabelValues(backupSchedule).Inc()
 	}
 }
+
+// RegisterBackupSeconds records the number of seconds a backup took.
+func (m *ServerMetrics) RegisterBackupSeconds(backupSchedule string, seconds float64) {
```

Comment:
suggest naming this method `RegisterBackupDuration`.

Reply:
Done